task_18/toma.lobanova/src/test/java/com/hillel/linkedlist/CustomLinkedListTests.java

```diff
@@ -0,0 +1,174 @@
+package com.hillel.linkedlist;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.EmptyStackException;
+import java.util.Iterator;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+
+public class CustomLinkedListTests {
+  CustomLinkedList list;
+
+  @Before
+  public void setUp() {
+    list = new CustomLinkedList();
+  }
+
+  @Test
+  public void addTest() {
+    list.add(1);
+    assertThat(list.toString(), is(equalTo("[1]")));
+    for (int i = 2; i < 6; i++) {
+      list.add(i);
+    }
+    assertThat(list.toString(), is(equalTo("[1, 2, 3, 4, 5]")));
+  }
+
+  @Test
+  public void iteratorTestEmptyList() {
+    StringBuffer string = new StringBuffer();
+    for (Object element : list) {
+      string.append(element);
+    }
+    assertThat(string.toString(), is(""));
+    Iterator iterator = list.iterator();
+    string = iterate(iterator);
+    assertThat(string.toString(), is(""));
+  }
+
+  private void addList() {
+    for (int i = 1; i < 10; i++) {
+      list.add(i);
+    }
+  }
+
+  private StringBuffer iterate(Iterator iterator) {
+    StringBuffer string = new StringBuffer();
```

Comment:
StringBuffer is slower than StringBuilder. If you don't work in multithreaded program it is better to use the second one.